flink/v1.14/flink/src/main/java/com/netease/arctic/flink/table/ArcticDynamicSource.java

```diff
@@ -322,12 +322,26 @@ protected Schema getProjectedSchema() {
       LOG.info("projectFields is null.");
       projectedSchema = arcticTable.schema();
     } else {
+      if (arcticTable.isUnkeyedTable()) {
+        throw new UnsupportedOperationException(
+            "Unkeyed table doesn't support lookup join.");
+      }
+      List<String> primaryKeys = arcticTable.asKeyedTable().primaryKeySpec().fieldNames();
+      List<Integer> projectFieldList = Arrays.stream(projectFields).boxed().collect(Collectors.toList());
+      List<Types.NestedField> columns = arcticTableSchema.columns();
+      for (int i = 0; i < arcticTableSchema.columns().size(); i++) {
+        if (primaryKeys.contains(columns.get(i).name()) && !projectFieldList.contains(i)) {
+          projectFieldList.add(i);
+          LOG.info(
+              "add identifier field [{}] to projected schema, due to this field is mismatched.",
```

Comment:
Pay attention to the grammar format of log messages, and the first letter should be capitalized.